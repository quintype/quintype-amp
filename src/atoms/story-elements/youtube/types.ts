import { Common } from "../../../atoms/common-types";
import { Config } from "../../../types/config";
import { Story, StoryElement } from "../../../types/story";

export interface YoutubeElementTypes {
  story: Story;
  config: Config;
  element: StoryElement;
}

export interface DefaultYoutubeElementTypes extends Common {
  element: StoryElement;
}
