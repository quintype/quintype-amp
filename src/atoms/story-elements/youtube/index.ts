import { YouTube } from "./youtube";
export { YouTube };
