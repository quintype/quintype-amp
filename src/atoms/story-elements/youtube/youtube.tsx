import React, { Fragment } from "react";
import { StoryElementProps } from "../types";
import getYouTubeID from "@rvgpl/get-youtube-id";
import { Common } from "../../common-types";
import Helmet from "react-helmet";
import styled from "styled-components";

const StyledYouTube = styled.div.attrs(({ inlineStyles }: StyledYouTubeTypes) => ({
  style: inlineStyles
}))<StyledYouTubeTypes>``;

export interface StyledYouTubeTypes {
  inlineStyles?: object;
}

type YouTubeProps = StoryElementProps & Common;

const YouTube = ({ element, layout = "responsive", width = "480", height = "270", inlineStyles }: YouTubeProps) => {
  const youtubeID = element.url && getYouTubeID(element.url);

  if (!youtubeID) {
    return null;
  }

  return (
    <Fragment>
      <Helmet>
        <script async={undefined} custom-element="amp-youtube" src="https://cdn.ampproject.org/v0/amp-youtube-0.1.js" />
      </Helmet>
      <StyledYouTube inlineStyles={inlineStyles}>
        <amp-youtube data-videoid={youtubeID} layout={layout} width={width} height={height} />
      </StyledYouTube>
    </Fragment>
  );
  // return
};

export { YouTube };
