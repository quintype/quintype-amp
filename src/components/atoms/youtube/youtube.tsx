import React, { Fragment } from 'react'
import { Helmet } from 'react-helmet'
import {YoutubeTypes} from './types'

const Youtube = (props: YoutubeTypes) => {
  return (
    <Fragment>
      <Helmet>
        <script async={undefined} custom-element="amp-youtube" src="https://cdn.ampproject.org/v0/amp-youtube-0.1.js"></script>
        <meta name="lorem ipsum" content="YOUTUBE script added" />
      </Helmet>
      <amp-youtube {...props}>{props.children}</amp-youtube>
    </Fragment>
  )
}

export default Youtube
