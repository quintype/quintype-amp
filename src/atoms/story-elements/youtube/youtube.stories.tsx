import React from "react";
import { storiesOf } from "@storybook/react";
import { YouTube } from "./youtube";
import { textStory, config } from "../../../__fixtures__";
import { Layout } from "../../layout";

const sampleYouTubeElement = {
  description: "",
  "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/2a3f1e74-df4b-4c8e-99de-57a69366b77f",
  type: "youtube-video",
  "family-id": "8256f5d1-5703-4c3f-8fc0-a6c4f44e531c",
  title: "",
  id: "2a3f1e74-df4b-4c8e-99de-57a69366b77f",
  url: "https://www.youtube.com/watch?v=jlEbFvGwcUc",
  "embed-url": "https://www.youtube.com/embed/jlEbFvGwcUc",
  metadata: {},
  subtype: null
};

storiesOf("Youtube", module)
  .addDecorator((story) => (
    <Layout story={textStory} config={config}>
      {story()}
    </Layout>
  ))
  .add("Default", () => <YouTube element={sampleYouTubeElement} />)
  .add("with custom styles", () => (
    <YouTube
      element={sampleYouTubeElement}
      inlineStyles={{
        border: "2px solid red"
      }}
    />
  ));
