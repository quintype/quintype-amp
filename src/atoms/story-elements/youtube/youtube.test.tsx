import React from "react";
import { YouTube } from "./youtube";
import { shallow } from "enzyme";

const sampleYouTubeElement = {
  description: "",
  "page-url": "/story/05066b94-784b-4913-971b-c01a1bcfb43d/element/2a3f1e74-df4b-4c8e-99de-57a69366b77f",
  type: "youtube-video",
  "family-id": "8256f5d1-5703-4c3f-8fc0-a6c4f44e531c",
  title: "",
  id: "2a3f1e74-df4b-4c8e-99de-57a69366b77f",
  url: "https://www.youtube.com/watch?v=jlEbFvGwcUc",
  "embed-url": "https://www.youtube.com/embed/jlEbFvGwcUc",
  metadata: {},
  subtype: null
};

describe("Youtube", () => {
  it("should render default", () => {
    const wrapper = shallow(<YouTube element={sampleYouTubeElement} />);
    expect(wrapper).toMatchSnapshot();
  });
});
