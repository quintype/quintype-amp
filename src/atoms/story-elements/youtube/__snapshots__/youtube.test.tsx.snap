// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Youtube should render default 1`] = `
<Fragment>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <script
      custom-element="amp-youtube"
      src="https://cdn.ampproject.org/v0/amp-youtube-0.1.js"
    />
  </HelmetWrapper>
  <amp-youtube
    amp-access="loggedIn"
    data-videoid="jlEbFvGwcUc"
    height="270"
    layout="responsive"
    width="480"
  />
</Fragment>
`;
